### DISCUSSION TITLE
PlotlyChart

### DISCUSSION THREAD

**Discussion by mogeebqaid** [flet-dev/flet#3747] [Ideas]
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the requested feature

I tried using the Plotly chart, but it seems that the interactivity feature is not working.
Can this issue be solved to make it possible to fully integrate Plotly with flet?

Code example to reproduce the issue:

```python
import plotly.graph_objects as go
import plotly.express as px
import pandas as pd
import flet as ft
from flet.plotly_chart import PlotlyChart
from flet import WebView

def main(page: ft.Page):
    df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/finance-charts-apple.csv')

    fig = px.line(df, x='Date', y='AAPL.High', title='Time Series with Range Slider and Selectors')

    fig.update_xaxes(
        rangeslider_visible=True,
        rangeselector=dict(
            buttons=list([
                dict(count=1, label="1m", step="month", stepmode="backward"),
                dict(count=6, label="6m", step="month", stepmode="backward"),
                dict(count=1, label="YTD", step="year", stepmode="todate"),
                dict(count=1, label="1y", step="year", stepmode="backward"),
                dict(step="all")
            ])
        )
    )
    page.add(PlotlyChart(fig, expand=True))

ft.app(main)
```

**Comment by ndonkoHenri** [comment]
Matplotlib and Plotly charts are static in Flet. Internally, they just render the provided figures as images.
For interactive charts atm, see other chart types we have: https://flet.dev/docs/controls/charts

**Reply by wendo298moraes** [reply]
When I need a graph that is based on a spreadsheet, isn't building it statically a better solution? I'm in a project and would like an opinion on how to build this dash

**Comment by manjeetparjapati** [comment]
Yes, you can use the WebView control in Flet to display an interactive Plotly chart. The WebView allows you to embed HTML content, including Plotly plot figures, making it a great option for full interactivity. However, please note that the WebView control currently supports iOS and Android only, so this solution won't work on Windows or other platforms.

If you're targeting those supported platforms, you can generate the Plotly chart as an HTML string (using plotly.io.to_html) and pass it to the WebView control for display.

**Comment by pychans** [comment]
Looking forward to webvie(desktop and browser versions ), which can solve this big problem

**Comment by inspectorG4dget** [comment]
Any update on this at all? I really need to be able to display an interactive sankey diagram (created with plotly) in my flet app

**Comment by FeodorFitsner** [comment]
We have interactive Matplotlib charts. Plotly is not an interactive - it's just an image.
